### PR TITLE
`azurerm_postgresql_flexible_server` - move the error check for `storage_mb` from `CustomizeDiff` to `Create/Update` func

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description
I run `tf apply` against below tf config 1 and the resources are created successfully. Then I update `name` of `azurerm_postgresql_flexible_server` and change `storage_mb` from 65536 to 32768 like below tf config 2 and run `tf apply` again. I expect that TF would re-create the resource successfully but it failed with below error. What is happening is that the validation in the customized diff is catching the error because the validation doesn't know that it is going to be destroyed and re-created. So [this error check](https://github.com/hashicorp/terraform-provider-azurerm/blob/33d93df64ce1386ac1eec39afa500f8b185f26d8/internal/services/postgres/postgresql_flexible_server_resource.go#L388) would always check if new storage_mb is lower than old storage_mb. Hence, we have to move the error check for `storage_mb` from `CustomizeDiff` to `Create/Update` func.
 
Error Message:
```
Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: 1 error occurred:
│       * 'storage_mb' can only be scaled up, expected the new 'storage_mb' value (32768) to be larger than the previous 'storage_mb' value (65536)
│
│
│
│   with azurerm_postgresql_flexible_server.test,
│   on main.tf line 23, in resource "azurerm_postgresql_flexible_server" "test":
│   23: resource "azurerm_postgresql_flexible_server" "test" {
```
 
The tf config 1 for the first apply:
```
provider "azurerm" {
  features {}
}
 
resource "azurerm_resource_group" "test" {
  name     = "acctestRG-postgresql-test01373"
  location = "eastus"
}
 
resource "azurerm_postgresql_flexible_server" "test" {
  name                   = "acctest-fs-test01373"
  resource_group_name    = azurerm_resource_group.test.name
  location               = azurerm_resource_group.test.location
  administrator_login    = "adminTerraform"
  administrator_password = "QAZwsx123"
  sku_name               = "GP_Standard_D2s_v3"
  version                = "12"
  zone                   = 2
  storage_tier           = "P6"
  storage_mb             = 65536
}
```
 
The tf config 2 for the second apply:
```
provider "azurerm" {
  features {}
}
 
resource "azurerm_resource_group" "test" {
  name     = "acctestRG-postgresql-test01373"
  location = "eastus"
}
 
resource "azurerm_postgresql_flexible_server" "test" {
  name                   = "acctest-newfs-test01373"
  resource_group_name    = azurerm_resource_group.test.name
  location               = azurerm_resource_group.test.location
  administrator_login    = "adminTerraform"
  administrator_password = "QAZwsx123"
  sku_name               = "GP_Standard_D2s_v3"
  version                = "12"
  zone                   = 2
  storage_tier           = "P6"
  storage_mb             = 32768
}
```

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

image
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_postgresql_flexible_server` - move the error check for `storage_mb` from `CustomizeDiff` to `Create/Update` func

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/postgres/postgresql_flexible_server_resource.go [hunk 1]
```diff
@@ -384,11 +384,6 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			newMb = newStorageMbRaw.(int)
 			newTier = newTierRaw.(string)
 
-			// storage_mb can only be scaled up...
```
Comment (katbyte) [inline internal/services/postgres/postgresql_flexible_server_resource.go:387]: is it not possible to also detect that something has changed forcing a new resoruce to be created and just not throw the error?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/postgres/postgresql_flexible_server_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/postgres/postgresql_flexible_server_resource.go:387]: Unfortuanately, I didn't find any way to ask terraform if the resource is going to be destroyed or not, I assume that bit is only known by the core runtime. So I have to move it to create/update func.

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for the comment. Please take another look.

File: internal/services/postgres/postgresql_flexible_server_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/postgres/postgresql_flexible_server_resource.go:387]: 👍

**Review by katbyte — APPROVED** [review]
LGTM 🌻

**Pull request merged**